### PR TITLE
static checks

### PR DESCRIPTION
**Pull request by jonesmac**

### CODE REVIEW

File: packages/ts-scripts-yarn3/src/actions/statics.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  constructor(output: string, dependency: string) {
+    this.dependency = dependency
+    this.dependencyEntries = this.formatOutput(output)
+  }
+
+  public detect() {
+    const result = this.dependencyEntries.reduce(this.detectReducer, this.resultsFactory(this.dependency))
+    if (result.duplicateVersions.length) {
+      console.error(`\n🚨 Duplicates found for: ${this.dependency} \n`)
```
Comment (JoelBCarter) [inline packages/ts-scripts-yarn3/src/actions/statics.ts:34]: I think this is where you'd want to use `EOL` from `os`.  When you're outputting text.  I suspect all consoles handle it equally at this point

File: packages/ts-scripts-yarn3/src/actions/statics.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  private resultsFactory = (dependency: string): Results => ({ currentVersion: undefined, dependency, duplicateVersions: [] })
+}
+
+export const detectDuplicates = (depsFromPackageJSON?: string[]) => {
+  let exitCode = 0
+  let dependencies: string[]
+  return safeExit(() => {
+    if (depsFromPackageJSON === undefined || depsFromPackageJSON.length === 0) {
```
Comment (JoelBCarter) [inline packages/ts-scripts-yarn3/src/actions/statics.ts:71]:
```suggestion
    if (depsFromPackageJSON?.length === 0) {
```

File: packages/ts-scripts-yarn3/src/actions/statics.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  })
+}
+
+export const confirmStaticPackages = () => {
+  console.log(chalk.green('Confirming Static Packages'))
+
+  const pathToPackageJSON = `${process.cwd()}/package.json`
+  const packageJSON = readFileSync(pathToPackageJSON).toString()
+  const parsedPackageJSON = JSON.parse(packageJSON)
+  const statics = parsedPackageJSON?.xy?.deps.statics
```
Comment (JoelBCarter) [inline packages/ts-scripts-yarn3/src/actions/statics.ts:110]:
```suggestion
  const statics = parsedPackageJSON?.xy?.deps?.statics
```

File: packages/ts-scripts-yarn3/src/actions/statics.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    return exitCode
+  })
+}
+
+export const confirmStaticPackages = () => {
+  console.log(chalk.green('Confirming Static Packages'))
+
+  const pathToPackageJSON = `${process.cwd()}/package.json`
+  const packageJSON = readFileSync(pathToPackageJSON).toString()
+  const parsedPackageJSON = JSON.parse(packageJSON)
```
Comment (JoelBCarter) [inline packages/ts-scripts-yarn3/src/actions/statics.ts:109]: I think `package.json` is already parsed and in the ENV if you're running under `yarn`.  I don't know if we are though depending on all the `execSync`.  Might be advantage to using the pre-parsed one?

Comment (jonesmac) [reply packages/ts-scripts-yarn3/src/actions/statics.ts:109]: I did a cursory look for some place that had the parsed package.json but didn't find it.  All I saw was some $npm... variables that took known values from package.json like `versions`.  If you can point me to the fully parsed ENV variable I will bow to your google skillz

**Pull request merged**